### ISSUE TITLE
`prefer_function_declarations_over_variables` is reported for final fields

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3450]
```dart
class A {
  final Function a = () {}; // there is a lint here

  Function b = () {};

  var c = () {};

  final d = () {}; // there is a lint here
}
```

Currently, there is a [lint](https://dart-lang.github.io/linter/lints/prefer_function_declarations_over_variables.html) for `a` and `d`, however I don't understand why it is only reported for `final` fields.

**Comment by srawlins** [comment]
See https://github.com/dart-lang/linter/issues/823 for some context.

**Comment by asashour** [comment]
Thanks a lot.
